std/traits.d

```diff
@@ -6490,6 +6509,38 @@ unittest
     }
 }
 
+/**
+Returns the number of full binary digits of precision representable by the scalar type `T`.
+ */
+template precision(T)
+    if (isScalarType!T)
+{
+    static if (isFloatingPoint!T)
+        enum int precision = T.mant_dig;
```

Comment:
```D
static assert(is(typeof(real.mant_dig) == int));
```

Also, D's integer promotion rules generally make `int` the safest and most user-friendly type when the value is guaranteed to be both small and positive.